docs/contributing-overview.md

```diff
@@ -0,0 +1,74 @@
+---
+id: contributing-overview
+title: Contributing to Single-spa
+sidebar_label: Overview
+---
+
+Thanks for checking out single-spa! We're excited to hear and learn from you.
+
+We've put together the following guidelines to help you figure out where you can best be helpful.
+
+## Table of Contents
+
+0. [Types of contributions we're looking for](#types-of-contributions-were-looking-for)
+0. [Ground rules & expectations](#ground-rules-expectations)
+0. [How to contribute](#how-to-contribute)
+0. [Style Guide](#style-guide)
+0. [Setting up your environment](#setting-up-your-environment)
+0. [Contribution review process](#contribution-review-process)
+0. [Community](#community)
+
+## Types of contributions we're looking for
+
+There are many ways you can directly contribute to the guides (in descending order of need):
+
+* Examples
+* Helper Libraries (like single-spa-react) for missing frameworks
+* Bug fixes
+* Answering questions in the slack channel
+* Entirely new helper projects like [single-spa-angular-cli](https://github.com/PlaceMe-SAS/single-spa-angular-cli)
+
+Interested in making a contribution? Read on!
+
+## Ground rules & expectations
+
+Before we get started, here are a few things we expect from you (and that you should expect from others):
+
+* Be kind and thoughtful in your conversations around this project. We all come from different backgrounds and projects, which means we likely have different perspectives on "how open source is done." Try to listen to others rather than convince them that your way is correct.
+* Please read the single-spa [Contributor Code of Conduct](./CODE_OF_CONDUCT.md). By participating in this project, you agree to abide by its terms.
+* If you open a pull request, please ensure that your contribution passes all tests. If there are test failures, you will need to address them before we can merge your contribution.
+* When adding content, please consider if it is widely valuable. Please don't add references or links to things you or your employer have created as others will do so if they appreciate it.
+
+## How to contribute
+
+If you'd like to contribute, start by searching through the [issues](https://github.com/CanopyTax/single-spa/issues) and [pull requests](https://github.com/CanopyTax/single-spa/pulls) to see whether someone else has raised a similar idea or question.
+
+If you don't see your idea listed, and you think it fits into the goals of this guide, do one of the following:
+
+* **If your contribution is minor,** such as a small typo or bug fix, open a pull request.
+* **If your contribution is major,** such as a new feature, start by opening an issue first. That way, other people can weigh in on the discussion before you do any work.
+
+## Style guide
```

Comment:
This entire section is false. Joel and I both view coding as an art form and don't strictly follow any particular style guide, If we're following it it's accidental. Please remove this section.

Reply:
I should say that _I_ view it that way, and AFAIK so does Joel. 

Either way I'd like to see this entire section about Style guide being removed, because I hate the idea of strict style guides.

Reply:
Got it! Just took this from the example from Github. I agree with you!